### ISSUE TITLE
Exposing map of all units

### ISSUE THREAD

**Issue by Hunter522** [LLNL/units#106]
I have a use case where I'd like to present a user with a list of units to convert from via a dropdown menu and a second dropdown menu with compatible units to convert to.

It looks like `static const smap base_unit_vals` in `units.cpp` provides all the available units and their string representation in this library which is awesome. Problem is this is `static` and I need this exposed. As a workaround I just externed it in `units.hpp`:

```cpp
...
#include <unordered_map>
...
namespace units {
...
using smap = std::unordered_map<std::string, precise_unit>;
extern const smap base_unit_vals;
```

And remove the static specifier in `units.cpp`:

```cpp
const smap base_unit_vals{
```

**Comment by phlptp** [comment]
I had thought about this before.  In general I didn't want to include the <unordered_map> header in units.hpp and having that symbol be exposed would lead to some complications in other libraries I use units in that generate shared libraries.  

What I had thought about was adding a little wrapper in the detail::testing namespace defined at the bottom of the units.hpp file that just returned a const reference to some of the main maps, but only if the `ENABLE_UNIT_TESTING` or some other compile time define was enabled.   I will probably need this at some point for additional testing.

**Comment by Hunter522** [comment]
Yeah that was one drawback I noticed with including `<unordered_map>`, unnecessarily bloating your include file. Though I just now noticed that `std::unordered_map` is already made available from this include tree:

* `"units.hpp"` 
  * `"units_definitions.hpp"`
    * `"units_decl.hpp"`
      * `<functional>`
        * `<unordered_map>` tho only if `__cplusplus > 201402L`...or C++14 which I am using C++17

So `<unordered_map>` in `units.hpp` is actually not necessary for compilers >= C++14. Though if you're trying to target >=C++11 then I'm guessing you'd have to include it. I see no way to forward declare `std::unordered_map` [1]

As for the symbol conflict, I like your idea of having a separate const reference. I tried that and seemed to work fine:

`units.hpp`:

```cpp
namespace units {
...
using smap = std::unordered_map<std::string, precise_unit>;
extern const smap& base_unit_vals_ref;
```

`units.cpp`:

```cpp
static const smap base_unit_vals{
...
}

const smap& base_unit_vals_ref = base_unit_vals;
```

[1] https://blog.magnum.graphics/backstage/forward-declaring-stl-container-types/
